### PR TITLE
sched: fix nxsched_suspend_scheduler regression

### PR DESCRIPTION
**Pull request by hujun260**
## Summary
sched: fix nxsched_suspend_scheduler regression
This commit fixes the regression from https://github.com/apache/nuttx/pull/13877

## Impact
sched

## Testing
ci

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: sched/sched/sched_suspendscheduler.c [hunk 1]
```diff
@@ -63,7 +63,7 @@ void nxsched_suspend_scheduler(FAR struct tcb_s *tcb)
 {
   /* Handle the task exiting case */
 
-  if (tcb != NULL)
+  if (tcb == NULL)
```
Comment (anchao) [inline sched/sched/sched_suspendscheduler.c:66]: It is impossible for the caller of this function to pass an invalid tcb, right? let us remove this check.

**Review by anchao — CHANGES_REQUESTED** [review]

File: sched/sched/sched_suspendscheduler.c (hunk 1 above)
Comment (xiaoxiang781216) [reply sched/sched/sched_suspendscheduler.c:66]: pthread_terminate/pthread_exit/task_exit may pass NULL tcb, so it's better to do the check here instead spreading in many place.

Comment (hujun260) [reply sched/sched/sched_suspendscheduler.c:66]: In the latest implementation, we will call nxsched_suspend_scheduler(g_running_tasks[this_cpu()]); and g_running_tasks[this_cpu()] will be set to NULL in up_exit

Comment (anchao) [reply sched/sched/sched_suspendscheduler.c:66]: `nxsched_resume_scheduler()` and `nxsched_suspend_scheduler()` are paired functions. If the code resumes but skips suspend, the format in the trace will be wrong, which is problematic for backend IDEs such as system view/perfetto.

Comment (anchao) [reply sched/sched/sched_suspendscheduler.c:66]: @XuNeo cc

Comment (jasonbu) [reply sched/sched/sched_suspendscheduler.c:66]: @anchao this patch only fix the regression on !tcb -> tcb == NULL. we will try to make mempool compatible with relay free to allow when exit, reuse tcb after exit. we should keep the nxsched_suspend_scheduler no regression at higher priority.

Comment (anchao) [reply sched/sched/sched_suspendscheduler.c:66]: I prefer to remove the check here，and move the relative logic into arch exit, which hard to implement?

Comment (GUIDINGLI) [reply sched/sched/sched_suspendscheduler.c:66]: Next patch, we will modify the exit free logic, and remove the "if" check here.

Comment (GUIDINGLI) [reply sched/sched/sched_suspendscheduler.c:66]: @anchao

**Review by anchao — APPROVED** [review]

**Pull request merged**